terraform-tests/dynamodb_namespace_test.go

```diff
@@ -0,0 +1,45 @@
+package terraformtests
+
+import (
+	"path"
+
+	tfjson "github.com/hashicorp/terraform-json"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	. "csbbrokerpakaws/terraform-tests/helpers"
+)
+
+var _ = Describe("dynamodb-namespace", Label("dynamodb-ns-terraform"), Ordered, func() {
+	var (
+		plan                  tfjson.Plan
+		terraformProvisionDir string
+	)
+
+	defaultVars := map[string]any{
```

Comment:
From the [Ginkgo docs](https://onsi.github.io/ginkgo/#extracting-common-setup-beforeeach):
> Information is shared between closures via closure variables. It is idiomatic for these closure variables to be declared within the container node closure and initialized in the setup node closure. Doing so ensures that each spec has a pristine, correctly initialized, copy of the shared variable.

Initialising a non-const outside of a BeforeEach() can lead to subtle issues that are hard to debug as if one testcase alters the value, another testcase will the the altered value and not the original (expected) value. Because testcases run in random order, these issues take ages to debug.

Reply:
On seconds thoughts, this is an ordered test with a BeforeAll() rather than a BeforeEach(), so while the above point is true, the risk is lower as the tests are not randomised.